lib/hasPermissions.ts

```diff
@@ -170,55 +175,43 @@ export class HasPermissionsDirectiveVisitor<
     ? debugGetErrorMessage
     : prodGetErrorMessage;
 
-  public visitObject(object: GraphQLObjectType | GraphQLInterfaceType): void {
-    Object.values(object.getFields()).forEach(field => {
-      this.visitFieldDefinition(field);
-    });
-  }
-
-  public visitFieldDefinition(field: GraphQLField<unknown, TContext>): void {
-    const { resolve = defaultFieldResolver } = field;
-    /* istanbul ignore next (directives that do not declare a default policy) */
-    const { permissions, policy = defaultPolicy } = this.args;
-    if (!permissions || !permissions.length) {
-      return;
-    }
+  public getValidationForArgs(): ValidateFunction<TContext> | undefined {
+    const { permissions } = this.args;
     const cacheKey = JSON.stringify(Array.from(permissions).sort());
 
-    const { getErrorMessage } = this;
+    const hasPermissionsValidateFunction: ValidateFunction<TContext> = (
+      value: unknown,
+      _: unknown,
+      __: unknown,
+      context: TContext,
+      resolverInfo: AnyObject,
+    ): unknown => {
+      if (!permissions || !permissions.length) {
+        return value;
+      }
 
-    // eslint-disable-next-line no-param-reassign
-    field.resolve = function (
-      obj,
-      args,
-      context,
-      info: MissingPermissionsResolverInfo,
-    ): Promise<unknown> {
       const { checkMissingPermissions } = context;
       let missingPermissions = checkMissingPermissions.apply(this, [
         permissions,
         cacheKey,
-        obj,
-        args,
+        // TODO Luis: ValidateFunction doesn't receive field resolver's
+        // source and context maybe this will be necessary since I can't
+        // remove these from CheckMissingPermissions
```

Comment:
its important to do not break the API in here. we need to check if we use this feature in our projects. (access the obj and args)

Reply:
I would love to just remove all checkMissingPermissions arguments related to GraphQLFieldResolver, but since this project are public I suspect this isn't possible, even if we don't use it in our projects